### PR TITLE
Add tracing with OpenTelemetry

### PR DESCRIPTION
**Pull request by szh**
### Desired Outcome

Add support for OpenTelemetry tracing. This will allow more observability into the authentication routines which are what takes the longest in the secrets-provider process.

### Implemented Changes

- Added tracing statements to the Authenticate() workflow
- Reduced the file wait timeout from 1 second to 50 ms. This decreases the time spent waiting by 95%.

### Connected Issue/Story

CyberArk internal issue link: [ONYX-14605](https://ca-il-jira.il.cyber-ark.com:8443/browse/ONYX-14605)

#### Changelog

- [x] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage

- [x] This PR includes new unit and integration tests to go with the code
  changes, or
- [ ] The changes in this PR do not require tests

#### Documentation

- [ ] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]()
- [x] This PR does not require updating any documentation

#### Behavior

- [ ] This PR changes product behavior and has been reviewed by a PO, or
- [ ] These changes are part of a larger initiative that will be reviewed later, or
- [x] No behavior was changed with this PR

#### Security

- [ ] Security architect has reviewed the changes in this PR,
- [ ] These changes are part of a larger initiative with a separate security review, or
- [x] There are no security aspects to these changes

### CODE REVIEW

File: cmd/authenticator/main.go [hunk 1]
```diff
@@ -10,6 +11,7 @@ import (
 	"github.com/cyberark/conjur-authn-k8s-client/pkg/authenticator"
 	authnConfig "github.com/cyberark/conjur-authn-k8s-client/pkg/authenticator/config"
 	"github.com/cyberark/conjur-authn-k8s-client/pkg/log"
+	"github.com/cyberark/secrets-provider-for-k8s/pkg/trace"
```
Comment (szh) [inline cmd/authenticator/main.go:14]: Is it bad practice to have a dependency on the cyberark/secrets-provider-for-k8s repo, which has packages that depend on this repo?

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,8 +6,12 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+### Added
+- Add support for tracing with OpenTelemetry. This add a new function to the authenticator, `AuthenticateWithContext`. The existing funtion, `Authenticate()` is deprecated and will be removed in a future upddate. [cyberark/conjur-authn-k8s-client#423](https://github.com/cyberark/conjur-authn-k8s-client/pull/423)
```
Comment (diverdane) [inline CHANGELOG.md:10]: Teeny, tiny nit: Should be "This adds a new function".

**Review by diverdane — APPROVED** [review]
There's a typo on the CHANGELOG.md, but otherwise LGTM!!!

**Review by diverdane — APPROVED** [review]
LGTM!!!

**Pull request merged**